### PR TITLE
fix #145

### LINKED ISSUE #145: Pasting a number with decimal while decimal scale is set to 0 gives wrong formatted value
Setup
```jsx
<NumberFormat decimalScale={0} />
```
Paste 9.5 to it, formatted value should be 9. Currently it comes as 95 which is incorrect.

### PR DESCRIPTION
**Pull request by ad1992**
@s-yadav please look into this

### CODE REVIEW

File: src/number_format.js [hunk 1]
```diff
@@ -164,6 +164,10 @@ class NumberFormat extends React.Component {
   getNumberRegex(g: boolean, ignoreDecimalSeparator?: boolean) {
     const {format, decimalScale} = this.props;
     const {decimalSeparator} = this.getSeparators();
+
+    if(decimalScale === 0) {
```
Comment (s-yadav) [inline src/number_format.js]: This will break if the number has a thousand separators

**Comment by ad1992** [comment]
@s-yadav can you check now. I have made the fixes.

File: src/number_format.js [hunk 2]
```diff
@@ -590,9 +590,18 @@ class NumberFormat extends React.Component {
...
-    const {isAllowed} = props;
+    const {isAllowed, decimalScale} = props;
     const lastValue = state.value || '';
 
+    /*
+    * get the valid numerical values only before the decimal
+    * separator when decimal scale is 0, issue #145
+    */
+
+    if(decimalScale === 0) {
```
Comment (s-yadav) [inline src/number_format.js]: You should also check for if format prop is not defined otherwise it will break things.

**Comment by s-yadav** [comment]
looks good will merge.

**Pull request merged**